dom.go

```diff
@@ -1035,6 +1068,12 @@ func mount(pendingMounts ...Mounter) {
 		if mounter == nil {
 			continue
 		}
+		if c, ok := mounter.(Component); ok {
+			if c.Context().mounted {
+				continue
+			}
+			c.Context().mounted = true
```

Comment:
`unmount` below sets both `c.Context().unmounted = true` and `c.Context().mounted = false`, so we would probably need to set `unmounted = false` here too? Or alternatively address my comment about not having separate `mounted` and `unmounted` states above